test/extensions/filters/udp/dns_filter/dns_filter_test.cc

```diff
@@ -0,0 +1,93 @@
+#include "envoy/config/filter/udp/dns_filter/v2alpha/dns_filter.pb.h"
+#include "envoy/config/filter/udp/dns_filter/v2alpha/dns_filter.pb.validate.h"
+
+#include "common/common/logger.h"
+
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::AtLeast;
+using testing::ByMove;
+using testing::InSequence;
+using testing::Return;
+using testing::ReturnRef;
+using testing::SaveArg;
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+namespace {
+
+class DnsFilterTest : public testing::Test {
```

Comment:
Once you add the real code I would recommend having an integration test for this. It shouldn't be too hard to add looking at the UDP proxy example integration test.

Reply:
That's the plan!